app/src/main/java/io/github/cmput301f19t19/legendary_fiesta/ui/FriendsMoodsFragment.java

```diff
@@ -46,7 +53,67 @@ public View onCreateView(@NonNull LayoutInflater inflater,
 
         moodList = mView.findViewById(R.id.mood_list_friends);
 
-        moodList.setAdapter(moodEventAdapter);
+        moodFilter = mView.findViewById(R.id.filter_spinner_friends);
+
+        //When an item in the spinner is selected
+        moodFilter.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                filteredMoodList = new ArrayList<>();
+                chosenMoodType = -1;
+
+                //Assign the correct value to chosenMoodType depending on what the user has selected
+                switch (i){
+                    case 0: //Scared
+                        chosenMoodType = Mood.SCARED;
+                        break;
+                    case 1: //Happy
+                        chosenMoodType = Mood.HAPPY;
+                        break;
+                    case 2: //Surprised
+                        chosenMoodType = Mood.SURPRISED;
+                        break;
+                    case 3: //Sad
+                        chosenMoodType = Mood.SAD;
+                        break;
+                    case 4: //Angry
+                        chosenMoodType = Mood.ANGRY;
+                        break;
+                    case 5: //Disgusted
+                        chosenMoodType = Mood.DISGUSTED;
+                        break;
+                    case 6: //Neutral
+                        chosenMoodType = Mood.NEUTRAL;
+                        break;
+                    case 7: //None
```

Comment:
Minor nitpick, you can change this to

```java
                    default: //None
```

Also, because you have defined `chosenMoodType = -1;` before, you can safely change the whole block to:

```java
default:  // None
    break;
```

Reply:
Ohh okk